### ISSUE TITLE
[feature] Move Conan tests to pytest

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan#6527]
Hi!

As you know, I'm a pytest fan due its facility for testing. For example, when I write an unit test, I don't want to remember the method name for that specific test (e.g. assertNotIn), I just want to put `assert` and use a regular condition.

Also, sometimes I need to write a tool for my tests, like a small http server to mock my answer. Or let's say I'm running another python framework like Flask, and I need some tools for testing. Well, pytest offers +800 plugins, including Flask integration: https://plugincompat.herokuapp.com/

But you can say that there is big code base for testing in your project, and porting is not possible. No problem, pytest is able to run nose: https://docs.pytest.org/en/latest/existingtestsuite.html

In summary , pytest [helps you write better programs](https://docs.pytest.org/en/latest/#pytest-helps-you-write-better-programs)

This suggestion could be adopted for Conan 2.0

What I see to be done for moving Conan to pytest:

- Rename test files to `test_*.py`
- Rename test methods to `test_*()`
- Replace tools which are already covered by pytest plugins

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by jgsogo** [comment]
Count with my sword :wink: 

Right now we are starting to plan the transition to Conan 2.0, the compatibility mode and after that we should maintain two branches: something like an LTS for Conan 1.x and the `rc` for Conan 2.0. This is going to increase the complexity of our CI and test suite and moving to `pytest` cannot take precedence over the roadmap.

But... probably we can add a branch to `conan_ci_jenkins` using `pytest` (running this `develop` branch ) and start moving things and checking that everything works (and fails) the same way --> https://github.com/conan-io/conan_ci_jenkins/issues/15

**Comment by memsharded** [comment]
> Rename test files to test_*.py

I would say this is no longer necessary, from the docs:

> File names should start or end with “test”, as in test_example.py or example_test.py

That doesn't mean we shouldn't have a convention. Our codebase use so far xxxx_test.py. The truth is that in that way it is a bit less evident when tests are skipped because they don't match, but I think renaming all the files in the test suite is too much, isn't it?

**Comment by uilianries** [comment]
no problem with file names now, I think that was a limitation for older versions. The only limitation would be the method names, but now is fixed, so all good.

**Comment by jgsogo** [comment]
I'm a huge fan of having `test_module/test_xxxxxx.py` naming convention (yes, also for the directories). It is easier for me to realize which files belong to the test suite and which ones don't. At least, for the `unittests`, where I try to have one file/folder for each file in the codebase, with exactly the same structure and names. 

But I can agree with other convention, I've been creating `module/xxxxx_test.py` files in this Conan repository for a while and I'm still sane 😄

**Comment by memsharded** [comment]
Ok, lets move the filenames to ``test_xxxx.py``, because it is more evident what is a test and what not (we have missed running tests in the past because they were not named correctly. Would you like to shoot a PR with those changes @uilianries ?

For the folders, I am not convinced, I need to play with the IDE to understand why.

**Comment by czoido** [comment]
Closed via: https://github.com/conan-io/conan_ci_jenkins/pull/28
